src/Data/List/Properties.agda

```diff
@@ -122,28 +122,34 @@ map-injective finj {x ∷ xs} {y ∷ ys} eq =
 ------------------------------------------------------------------------
 -- mapMaybe
 
+length-catMaybes : ∀ xs → length (catMaybes {A = A} xs) ≤ length xs
+length-catMaybes []        = ≤-refl
+length-catMaybes (x  ∷ xs) = let ih = length-catMaybes xs
+  in maybe {B = λ x → length (catMaybes (x ∷ xs)) ≤ suc (length xs)}
```

Comment:
This is maybe just me but I find this a lot less readable.
It adds to the noise by forcing us to explicitly pass the motive `B` instead of
having the *assistant* write it for us.

Reply:
Yes, I agree with you! Will update.